Extension/src/common.ts

```diff
@@ -1304,3 +1304,16 @@ export function sequentialResolve<T>(items: T[], promiseBuilder: (item: T) => Pr
         return promiseBuilder(nextItem);
     }, Promise.resolve());
 }
+
+// Quote a file path if it includes space.
+export function quoteFilePaths(path: string): string {
+    // Check if the path is not already quoted.
+    if (!/".*?"/.test(path) && !/'.*?'/.test(path)) {
```

Comment:
* #7891 relates to use of backtick quoting (which is only supported on mac and linux), but I don't see any handling of backtick quoting here.
* It looks like you are using a pattern to check if the entire path is surrounded by quotes.  Quoted portions may be contained anywhere within an argument. If there is already quoting within an arg, placing quotes around the whole arg would requiring escaping those internal quotes.  Beware if detecting internal quotes within an arg, as they should not be considered as quoting if escaped (`\"`)
* It's unnecessary to detect if the arg contains a path, as args that contain paths are no different from any other arg contents with regards to whether or not they require quoting.  i.e. macros/defines may have spaces.  (So, this function should be something like `quoteArgument`).  Since it's perfectly acceptable to quote an argument that does not contain spaces, consider quoting ALL arguments (if not already fully quoted), and escaping any quoting within it.  That might be easier than detecting if quoting is necessary.  Backticks might be different - I suspect that must always fully quote an arg and cannot be escaped.
* Single-quotes are supported only on Linux.  They behave as literals on Windows, so shouldn't be checked for on Windows.

Reply:
based on our previous discussion:
Assumption: Backtick quoting must be around the entire argument. (Whereas single-quote and double-quote quoting can start in the middle of an argument. For example: -DTEST="TEST TEST", or -DTEST='TEST TEST' on Linux/Mac



The algorithm:



1. Trim the arg to remove leading and trailing spaces.
2. (On Linux/Mac) If completely enclosed within backtick quoting, leave it alone.
3. Check if the string contains any escaped double-quote characters. If so, leave the arg alone, as the user is doing the quoting themselves. Adding additional quoting would change the behavior. (i.e. -DTEST="TEST TEST" is different from "-DTEST=\"TEST TEST\"")
4. On Linux or Mac, check if the string contains any unescaped single-quote characters. If so, leave the arg alone. (same reason as above).
5. Scan for any spaces not preceded by a slash. Only if there is an un-escaped space is quoting needed.
6. If any un-escaped space is found, remove the "\" from any escaped spaces ("\ ", in case there were some as they will no longer be valid if quoted), and put double-quotes around the whole thing.